### PR TITLE
Card Improvements

### PR DESCRIPTION
**Pull request by nicknisi**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

This PR fixes the card implementation, simplifying its use by taking advantage of named child properties. 

This is part of the work for improving the card implementation in #978.

**Properties**

- `onAction?: () => void` - an optional action handler for the card content. This adds the primary action styles to the card when provided and is called back when the card's content is clicked on
- `mediaSrc?: string` - An optional path to an image that when provided will populate the media section within the card's main content area
- `mediaTitle?: string` - An optional string property that sets the media's `title` property and will be displayed when the media is hovered over
- `square?: boolean` - When provided, the square class will be applied, rendering the media with a 1:1 ratio
- `title?: string` - When provided adds a default and consistently-styled title section to the card's content
- `subtitle?: string` - When provided adds a default and consistently-styled subtitle section to the card's content

**Named Children**
All children are optional and have the type `() => RenderResult`.

- `header` - The header section goes above the main content section
- `content` - The main content section. If `mediaSrc`, `title`, or `subtitle` are provided, they will appear above the contents of this section
- `actionButtons` - The action buttons to render on the left side of the card
- `actionIcons` - The action icons to render on the right side of the card

**Example**
```tsx
<Card
	onAction={() => {}}
	mediaSrc={mediaSrc}
	title="Hello, World"
	subtitle="A pretty picture"
>
	{{
		header: () => <div>Header Content</div>,
		content: () => <p>Travel the world today.</p>,
		actionButtons: () => (
			<virtual>
				<Button>Read</Button>
				<Button>Bookmark</Button>
			</virtual>
		),
		actionIcons: () => (
			<virtual>
				<Icon type="secureIcon" />
				<Icon type="downIcon" />
				<Icon type="upIcon" />
			</virtual>
		)
	}}
</Card>
```

**Media Card**
image

**Card with actions**
image

**Card with header, media, content, and actions**
image

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-9rbnar0fr.now.sh
* Tests Deployment: https://dojowidgets-o2win0qeb.now.sh

**Comment by tomdye** [comment]
@nicknisi looks to me like theres some padding missing on the material action bar

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-ooyp2hh9v.now.sh
* Tests Deployment: https://dojowidgets-asmlntvju.now.sh

**Comment by nicknisi** [comment]
Good call. I added the appropriate padding to the material theme actions. 
image

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-73ussv3w1.now.sh
* Tests Deployment: https://dojowidgets-5hpj0wif6.now.sh

File: src/theme/material/card.m.css [hunk 1]
```diff
@@ -14,12 +16,53 @@
 	composes: mdc-card__action-icons from '@material/card/dist/mdc.card.css';
 }
 
+.primary {
+	composes: mdc-card__primary-action from '@material/card/dist/mdc.card.css';
+	/* padding: 1rem 1rem 0rem 1rem; */
```
Comment (tomdye) [inline src/theme/material/card.m.css]: comments

**Comment by nicknisi** [comment]
Thanks, @tomdye I pushed up a commit to remove the commit. It is ready to go from there!

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-odjco16nn.now.sh
* Tests Deployment: https://dojowidgets-bewli7s1z.now.sh

**Pull request merged**